sale_order_type/models/sale.py

```diff
@@ -67,41 +75,70 @@ def _compute_sale_type_id(self):
                 sale_type = record._default_type_id()
             record.type_id = sale_type
 
-    @api.onchange("type_id")
-    def onchange_type_id(self):
-        # TODO: To be changed to computed stored readonly=False if possible in v14?
-        vals = {}
+    @api.depends("type_id")
+    def _compute_warehouse_id(self):
+        res = super()._compute_warehouse_id()
         for order in self:
             order_type = order.type_id
-            # Order values
-            vals = {}
             if order_type.warehouse_id:
-                vals.update({"warehouse_id": order_type.warehouse_id})
+                order.warehouse_id = order_type.warehouse_id
+        return res
+
+    @api.depends("type_id")
+    def _compute_picking_policy(self):
+        if hasattr(super(), "_compute_picking_policy"):
+            return super()._compute_picking_policy()
```

Comment:
```suggestion
            res = super()._compute_picking_policy()
```

and `return res` at the end.